rpc/src/rpc.rs

```diff
@@ -3133,6 +3219,20 @@ pub mod rpc_accounts {
             meta.get_program_accounts(&program_id, config, filters, with_context)
         }
 
+        fn get_secondary_index_key_size(
+            &self,
+            meta: Self::Metadata,
+            pubkey_str: String,
+            commitment: Option<CommitmentConfig>,
```

Comment:
let's move this arg into a config struct in anticipation of wanting other args in the future (eg. min_context_slot, etc)

Reply:
Should I reuse the existiong "RpcContextConfig" which adds min_context_slot then? Or rather a fully new one that can be expanded. There are a lot of existing config types, would rather just re-use one that fits than creating a new one I dont actually currently have need for. Also if I add min_context_slot, I'll need to handle it as well.

Reply:
> Should I reuse the existiong "RpcContextConfig"

not for the time being. i'd just add a new method-specific config type for now (ala. `RpcTransactionConfig`, `RpcBlockConfigWrapper`, etc). the sole field can be the optional commitment declared like it is here.  idea behind the per-method-config is to ease future extension (eg. by adding an `RpcContextConfig`), should that be deemed necessary.

Reply:
Sorry to yank you back and forth here, but it's not necessary to add the method-specific config right now.
Because of the way struct-to-map serialization happens, and the way we handle (flatten) CommitmentConfig and other fields, we can use:
- `Option<CommitmentConfig>`, which will serialize like `{"commitment": "<COMMITMENT_STR>"}`
- or, `Option<RpcContextConfig>`, which will serialize like `{"commitment": "<COMMITMENT_STR>", "minContextSlot":<SLOT>}`
- and later, `Option<RpcSecondaryIndexConfig>`, which could contain a flattened `RpcContextConfig` and would serialize like `{"commitment": "<COMMITMENT_STR>", "minContextSlot":<SLOT>, "someOtherField":<VALUE>}`
...all without breaking/backward-compatibility issues.

Imho it's not worth carrying around that extra struct until such time as it doesn't duplicate one that already exists.
I don't have a strong opinion about `Option<CommitmentConfig>` vs `Option<RpcContextConfig>` right now. I don't see a strong need for min_context_slot, but no real harm in having it.